internal/testutil/acc/atlas.go

```diff
@@ -0,0 +1,116 @@
+package acc
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"runtime"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/atlas-sdk/v20231115007/admin"
+)
+
+// TestMainExecution must be called from TestMain in the test package if ProjectIDExecution is going to be used.
+func TestMainExecution(m *testing.M) {
+	if !InUnitTest() {
+		atlasInfo.init = true
+		atlasInfo.resourceName = resourceName()
+	}
+
+	exitCode := m.Run()
+
+	if !InUnitTest() && atlasInfo.needsDeletion {
```

Comment:
reading this `if` makes me thinking that we're designing a non-trivial set of rules for which resources should be re-used or deleted. I see a potential risk here that tests could start _randomly_ failing due to concurrency issues and figuring what happened is hard.

Let me start first with:
1. Why this `if`? What is the rationale?
2. Could we make things simpler? e.g. re-use resource for every test file and make it just a serial process?
3. Is there an alternative way to write this?

Reply:
by default a project is created when you call ProjectIDExecution and it's deleted after all tests run.

i made a further optimization to allow reuse of a global project for that resource if defined so we avoid creating the project during the test run. 

i'm not sure the code can be simplified to fulfil those requirements, but we'll probably learn more as we use this in other resources and might be able to simplify

Reply:
i think in next PRs it'll be more clear how this is intended to be used, the thing with this PR is that there is more implementation details than real use of the feature.

Reply:
i've simplified the `if` condition a little bit, let me know if it helps